### PR TITLE
Changes to run the Beverly Hills verkle testnet

### PR DESCRIPTION
**Pull request by gballet**
The upcoming Beverly Hills testnet is using PoS and does not include the proofs in blocks. This PR includes the required changes for it to work, namely by resetting the block format to its old type.

This is still in draft, because the PR needs to be rebased on top of master, and also needs to store the proofs (who are still generated) into the database, so that they can be used by e.g. a verkle block explorer.

### CODE REVIEW

File: core/state/statedb.go [hunk 1]
```diff
@@ -158,6 +180,21 @@ func New(root common.Hash, db Database, snaps *snapshot.Tree) (*StateDB, error)
 	return sdb, nil
 }
 
+func (s *StateDB) Snaps() *snapshot.Tree {
```
Comment (gballet) [inline core/state/statedb.go:183]: move to its own file

File: miner/worker.go [hunk 2]
```diff
@@ -98,6 +98,8 @@ type environment struct {
 	txs      []*types.Transaction
 	receipts []*types.Receipt
 	uncles   map[common.Hash]*types.Header
+
+	preRoot common.Hash
```
Comment (gballet) [inline miner/worker.go:102]: move to Finalize

Comment (gballet) [reply miner/worker.go:102]: and assemble

**Comment by holiman** [comment]
This is draft and seem heavily bitrotted. I'll just assume it's not relevant any more

**Pull request closed without merging**